projects/plugins/wpcomsh/widgets/class-pd-top-rated.php

```diff
@@ -78,7 +79,7 @@ public function widget( $args, $instance ) {
 					}
 				}
 
-				if ( is_array( $current_category ) && (int) $current_category[0]->cat_ID > 0 ) {
+				if ( is_array( $current_category ) && count( $current_category ) > 0 && (int) $current_category[0]->cat_ID > 0 ) {
```

Comment:
Could simplify this, since a zero-count array is falsey.
```suggestion
				if ( is_array( $current_category ) && $current_category && (int) $current_category[0]->cat_ID > 0 ) {
```
OTOH, if there's a possibility that someone might have put different keys or objects into the array, maybe like this would be safer
```suggestion
				if ( is_array( $current_category ) && isset( $current_category[0]->cat_ID ) && (int) $current_category[0]->cat_ID > 0 ) {
```